### PR TITLE
introspection-app: example

### PR DESCRIPTION
**Pull request by oliver-wm**
A minimal example of native component being given
access to the guest vms ram region via a seL4SharedDataWithCaps
connection.

### CODE REVIEW

**Comment by oliver-wm** [comment]
@kent-mcleod could you please review.

File: apps/vm_introspect/tk1/devices.camkes [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+#define VM_RAM_SIZE 0x40000000
+#define VM_RAM_OFFSET 0x30000000
+#define VM_DTB_ADDR 0x82000000
+#define VM_INITRD_ADDR 0x80700000
+
+assembly {
+	composition {}
+	configuration {
+
+        vm0.memdev_paddr = VM_RAM_BASE;
```
Comment (WellMcGarnicle) [inline apps/vm_introspect/tk1/devices.camkes]: Does this need to take into account the offset?

Comment (oliver-wm) [reply apps/vm_introspect/tk1/devices.camkes]: Hi, can you confirm the `VM_RAM_BASE + VM_RAM_OFFSET` is the start of the guest memory?

Comment (WellMcGarnicle) [reply apps/vm_introspect/tk1/devices.camkes]: I may have limited context here, but using the offset seems to add up to the `VM_PADDR_RAM_BASE` a few lines above + the addresses in the untyped mmios below?

Comment (oliver-wm) [reply apps/vm_introspect/tk1/devices.camkes]: Makes sense. Thanks!

**Pull request closed without merging**